x-pack/plugins/session_view/public/components/ProcessTreeNode/index.tsx

```diff
@@ -31,18 +32,34 @@ interface ProcessDeps {
 export function ProcessTreeNode({
   process,
   isSessionLeader = false,
+  isOrphan,
   depth = 0,
   onProcessSelected,
 }: ProcessDeps) {
   const styles = useStyles({ depth });
+  const textRef = useRef<HTMLSpanElement>(null);
 
   const [childrenExpanded, setChildrenExpanded] = useState(isSessionLeader || process.autoExpand);
+  const { searchMatched } = process;
 
   useEffect(() => {
     setChildrenExpanded(isSessionLeader || process.autoExpand);
   }, [isSessionLeader, process.autoExpand]);
 
-  const event = process.getLatest();
+  useLayoutEffect(() => {
+    if (searchMatched !== null && textRef && textRef.current) {
+      const regex = new RegExp(searchMatched);
+
+      const text = textRef.current.innerText;
+      const html = text.replace(regex, (match) => {
+        return `<span style="${styles.searchHighlight}">${match}</span>`;
```

Comment:
would `match` be vulnerable to XSS?

not a big deal to address it now since this is a draft, but should be considered for later

Reply:
yea for sure... would be nice for a solution that used react without having to mod the dom this way. but yea, this search feature will probably change, so we can address later.